haystack_experimental/core/super_component/super_component.py

```diff
@@ -0,0 +1,420 @@
+        """
+        Process outputs from pipeline execution.
+
+        :param pipeline_outputs: Raw outputs from pipeline execution
+        :return: Dictionary of processed outputs
+        :raises InvalidMappingError: If output conflicts occur during auto-mapping
+        """
+        if output_mapping := self.output_mapping or self.pipeline.metadata.get("output_mapping"):
+            return self._map_explicit_outputs(pipeline_outputs, output_mapping)
+        else:
+            return self._map_auto_outputs(pipeline_outputs)
+
+    def _map_explicit_outputs(
+        self, pipeline_outputs: Dict[str, Dict[str, Any]], output_mapping: Dict[str, str]
+    ) -> Dict[str, Any]:
+        """
+        Map outputs according to explicit output mapping.
+
+        :param pipeline_outputs: Raw outputs from pipeline execution
+        :param output_mapping: Output mapping configuration
+        :return: Dictionary of mapped outputs
+        """
+        outputs: Dict[str, Any] = {}
+        for pipeline_output_path, wrapper_output_name in output_mapping.items():
+            comp_name, socket_name = self._split_component_path(pipeline_output_path)
+            if comp_name in pipeline_outputs and socket_name in pipeline_outputs[comp_name]:
+                outputs[wrapper_output_name] = pipeline_outputs[comp_name][socket_name]
+        return outputs
+
+    @staticmethod
+    def _map_auto_outputs(pipeline_outputs: Dict[str, Dict[str, Any]]) -> Dict[str, Any]:
+        """
+        Map outputs automatically based on matching names.
+
+        :param pipeline_outputs: Raw outputs from pipeline execution
+        :return: Dictionary of mapped outputs
+        :raises InvalidMappingError: If output conflicts occur during auto-mapping
+        """
+        outputs: Dict[str, Any] = {}
+        seen: set[str] = set()
+
+        for comp_output_dict in pipeline_outputs.values():
+            for socket_name, value in comp_output_dict.items():
+                if socket_name in seen:
+                    raise InvalidMappingError(
+                        f"Output conflict: output '{socket_name}' is produced by multiple components. "
+                        "Please provide an output_mapping to disambiguate."
+                    )
+                outputs[socket_name] = value
+                seen.add(socket_name)
+
+        return outputs
+
+    def _to_super_component_dict(self) -> Dict[str, Any]:
+        """
+        Convert to a SuperComponent dictionary representation.
+
+        :return: Dictionary containing serialized SuperComponent data
+        """
+        serialized_pipeline = self.pipeline.to_dict()
+        serialized = default_to_dict(
+            self,
+            pipeline=serialized_pipeline,
+            input_mapping=self.input_mapping, output_mapping=self.output_mapping
+        )
+        serialized["type"] = "haystack_experimental.core.super_component.super_component.SuperComponent"
```

Comment:
```suggestion
        serialized["type"] = generate_qualified_class_name(SuperComponent)
```
let's tie this to the class but dynamically reflect the path

this also needs to imported: `from haystack.core.serialization import generate_qualified_class_name`